core/editor/src/input/mouse.rs

```diff
@@ -1,5 +1,79 @@
 use bitflags::bitflags;
 
+// A position on the infinite canvas
+#[derive(Debug, Copy, Clone, Default, PartialEq)]
+pub struct DocumentPosition {
+	pub x: f64,
+	pub y: f64,
+}
+
+impl DocumentPosition {
+	pub fn distance(&self, other: &Self) -> f64 {
+		let x_diff = other.x - self.x;
+		let y_diff = other.y - self.y;
+		f64::sqrt(x_diff * x_diff + y_diff * y_diff)
+	}
+	pub fn rotate(&self, theta: f64) -> Self {
+		let cosine = theta.cos();
+		let sine = theta.sin();
+		Self {
+			x: self.x * cosine - self.y * sine,
+			y: self.x * sine + self.y * cosine,
+		}
+	}
+	pub fn add(&self, x: f64, y: f64) -> Self {
+		Self { x: self.x + x, y: self.y + y }
+	}
+	pub fn multiply(&self, x: f64) -> Self {
+		Self { x: self.x * x, y: self.y * x }
+	}
+}
+impl From<DocumentPosition> for (f64, f64) {
+	fn from(item: DocumentPosition) -> Self {
+		(item.x, item.y)
+	}
+}
+
+// The location of the viewport (or anything else) in the canvas
+#[derive(Debug, Copy, Clone, PartialEq)]
+pub struct DocumentTransform {
+	pub location: DocumentPosition,
+	pub radians: f64,
+	pub degrees: f64,
```

Comment:
I'm inclined to prefer degrees throughout the application since degrees are probably more useful in 2D graphics than radians. I'm probably neglecting to consider something so please push back if you disagree.

Reply:
Agree with @TrueDoctor s comment. For trigonometric functions and some other calculations radians usually are more natural  (lead to simpler equations), hence why sin() and cos() use radians. We can make sure however, that users only ever see degrees.